cpp/src/barretenberg/stdlib/primitives/bool/bool.cpp

```diff
@@ -397,7 +397,9 @@ void bool_t<ComposerContext>::assert_equal(const bool_t& rhs, std::string const&
     const bool_t lhs = *this;
     ComposerContext* ctx = lhs.get_context() ? lhs.get_context() : rhs.get_context();
 
-    if (lhs.is_constant() && rhs.is_constant()) {
```

Comment:
We want to put the composer in a failing state without changing existing code. We can't reuse the "both are witnesses" code  because the assert_equal that takes in values (ant not witness indices) only exists on the simulator.